### PR TITLE
feature/sof-4195 - ML Documentation Updates

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -252,13 +269,29 @@ nav:
...
+              # - Model Training
+              # - Model Prediction
+              # - Post-Processing
             - Units:                                    models-directory/machine-learning/units.md
+              # Not sure what we want to do with Data. I suggest that we have two headings:
+              # 1. Exabyte ML
+              # 2. Python ML (Might we hold off on this until AZ finishes the ESSE updates for model classes?)
             - Data:                                     models-directory/machine-learning/data.md
-            - Actions:                                  models-directory/machine-learning/actions.md
-            - Units:                                    models-directory/machine-learning/units.md
```
Comment (AcylSilane) [inline mkdocs.yml:260]: This was a duplicate page

File: mkdocs.yml [hunk 2]
```diff
@@ -110,9 +110,26 @@ nav:
             - Flags by Elemental Composition:        tutorials/templating/set-flag-by-composition.md
             - Magnetic Moment on Atoms by Specie:    tutorials/templating/set-magnetic-moment.md
         - Machine Learning (ML):
+              # Differentiate between Legacy and PythonML tutorials in Overview
             - Overview:                              tutorials/ml/overview.md
-            - Train ML Model:                        tutorials/ml/train-ml-model.md
-            - Predict New Properties:                tutorials/ml/predict-ml-properties.md
+            - Legacy:
```
Comment (timurbazhirov) [inline mkdocs.yml:115]: Legacy (ExabyteML)

File: mkdocs.yml [hunk 3]
```diff
@@ -252,13 +269,29 @@ nav:
...
+              # - Feature Engineering
+              # - Feature Selection (Already exists)
+              # - Hyperparameter Tuning
+              # - Model Training
+              # - Model Prediction
+              # - Post-Processing
             - Units:                                    models-directory/machine-learning/units.md
+              # Not sure what we want to do with Data. I suggest that we have two headings:
+              # 1. Exabyte ML
+              # 2. Python ML (Might we hold off on this until AZ finishes the ESSE updates for model classes?)
```
Comment (timurbazhirov) [inline mkdocs.yml:287]: There's no Python ML implementation yet

File: mkdocs.yml [hunk 4]
```diff
@@ -438,12 +481,14 @@ nav:
         - Components:
             - Overview:                                 workflows/components/overview.md
             - Subworkflows:                             workflows/components/subworkflows.md
+              # Add description of download_from_objet_storage unit
```
Comment (timurbazhirov) [inline mkdocs.yml:484]: Maybe just a quick mention

File: mkdocs.yml [hunk 5]
```diff
@@ -560,6 +605,7 @@ nav:
         - Data:
             - Overview:                                 properties/data/overview.md
             - Core:                                     properties/data/core.md
+              # Add example schema for new file_content property
```
Comment (timurbazhirov) [inline mkdocs.yml:608]: We should explain all IO unit types then

File: mkdocs.yml [hunk 6]
```diff
@@ -592,6 +638,8 @@ nav:
             - Phonon Dispersions:                       properties-directory/non-scalar/phonon-dispersions.md
             - Phonon Density of States:                 properties-directory/non-scalar/phonon-dos.md
             - Reaction Energy Profile:                  properties-directory/non-scalar/reaction-energy-profile.md
+            # Description of what the new File_Content property is, and what it does
+            - File Content :                            properties-directory/non-scalar/file_content.md
```
Comment (timurbazhirov) [inline mkdocs.yml:642]: Let's also add workflow as property here and link to it in explanation

File: mkdocs.yml (hunk 2 above)
Comment (AcylSilane) [reply mkdocs.yml:115]: Fixed

File: mkdocs.yml (hunk 3 above)
Comment (AcylSilane) [reply mkdocs.yml:287]: Adjusted to request no changes

File: mkdocs.yml (hunk 6 above)
Comment (AcylSilane) [reply mkdocs.yml:642]: Added as a page

File: mkdocs.yml (hunk 5 above)
Comment (AcylSilane) [reply mkdocs.yml:608]: Updated to clarify this

File: mkdocs.yml (hunk 4 above)
Comment (AcylSilane) [reply mkdocs.yml]: Updated to clarify this

File: mkdocs.yml [hunk 7]
```diff
@@ -110,9 +110,26 @@ nav:
             - Flags by Elemental Composition:        tutorials/templating/set-flag-by-composition.md
             - Magnetic Moment on Atoms by Specie:    tutorials/templating/set-magnetic-moment.md
         - Machine Learning (ML):
+              # Differentiate between Legacy and PythonML tutorials in Overview
             - Overview:                              tutorials/ml/overview.md
-            - Train ML Model:                        tutorials/ml/train-ml-model.md
-            - Predict New Properties:                tutorials/ml/predict-ml-properties.md
+            - Legacy (ExabyteML):
```
Comment (timurbazhirov) [inline mkdocs.yml:115]: Exabyte ML (Legacy)

Comment (AcylSilane) [reply mkdocs.yml]: Fixed in recent commit

**Comment by AcylSilane** [comment]
Merging per https://exabyte.atlassian.net/browse/SOF-4195?focusedCommentId=41292

**Pull request merged**